### DISCUSSION TITLE
Suggestions for improving the NWT Reading app

### DISCUSSION THREAD

**Discussion by AngelAcuna2000** [searchwork/nwt-reading-contrib#3] [Ideas]
Dear developers, 

I’m a user of your NWT Reading app, and I really enjoy it. It’s a great tool for studying the Bible and keeping track of my progress. I have some ideas for improving the app that I would like to share with you.

It would be great if the app could allow users to check off the chapters that they have read each day and also record the verse number that they left off at in each chapter. This way, users can easily resume their reading from where they stopped and keep track of their progress.

It would also be nice if the app could show users how many days they are behind, ahead, or if they are on schedule based on the reading speed that they selected. This would motivate users to catch up or maintain their reading pace and help them plan their reading time better.

Lastly, I think it would be helpful if the app could have a setting option that allows users to choose whether or not they want to check off all the previous chapters as read when they check off a later chapter. This way, users can either use this shortcut or avoid being asked every time if they want to mark all the previous chapters as read.

I hope you will consider these suggestions and let me know what you think. Thank you for your hard work and dedication to this project.

Sincerely, Angel Acuna

**Comment by wfleischer** [comment]
Thanks @AngelAcuna2000 for your suggestions! Really appreciate them.

> It would be great if the app could allow users to check off the chapters that they have read each day and also record the verse number that they left off at in each chapter. This way, users can easily resume their reading from where they stopped and keep track of their progress.

Personally I do this within the JW Library app. That is, I use the NWT Reading app to track my schedule, but use a bookmark within the JW Library app to track where I left within a reading section.

Do you find this practical or do you still think a feature within the NWT Reading app would be preferable?
 
> It would also be nice if the app could show users how many days they are behind, ahead, or if they are on schedule based on the reading speed that they selected. This would motivate users to catch up or maintain their reading pace and help them plan their reading time better.

Actually the app does display in a badge in the title how many days one is ahead or behind the reading schedule (see [here](https://play-lh.googleusercontent.com/oTO8sbYLC-sjpna5aVXz2V1peQ4BMEYn-6GwsbxNbAaU-7g-2rGQSH2FIkpDUjPIyg=w5120-h2880-rw) for instance).

> Lastly, I think it would be helpful if the app could have a setting option that allows users to choose whether or not they want to check off all the previous chapters as read when they check off a later chapter. This way, users can either use this shortcut or avoid being asked every time if they want to mark all the previous chapters as read.

The app does allow to check respectively uncheck anywhere in the schedule and will mark all previous sections as being read respectively all following as being unread. Is this what you are referring to?

Let me know if you have questions or further suggestions.

**Reply by AngelAcuna2000** [reply]
@wfleischer Thank you for your response, I really appreciate it! 

I have attached a video that demonstrates the issues I have encountered, which are the following:

- When I mark a passage as read, the app automatically marks all the previous passages as read too, even if I haven’t read them yet. I cannot undo this action or select individual passages to mark as read or unread.

- When I mark a passage as unread, the app also marks all the following passages as unread, even if I have already read them. I cannot prevent this from happening or choose which passages to mark as unread.

- The app does not always give the correct number of days that I am behind or ahead. This is because the app does not have a setting option to set a start date for my reading plan. The options “start over” and “Reset end date” do not solve this issue.

I have also attached a screenshot of another Bible reading tracker app that I used before. Unfortunately, this app is no longer available or compatible with my phone, which is why I switched to your app. The features that I really liked about this app are the following:

- The app shows which books of the Bible I have started reading by shading them with purple. The more purple there is, the more I have read from that book. If a book is fully purple, it means I have finished reading it. If a book is not shaded at all, it means I have not started reading it yet.

- The app allows me to tap on a book and see which chapters and verses I have read and which ones I still need to read. It also lets me enter and see the last verse I read in a chapter that I have not completed yet.

I would love to see something similar in NWT reading. For instance, it would be nice if it could display the progress of each book and passage with a color, shade, or percentage number so that I can easily see how much I have read without tapping on them. And when I do tap on them, instead of taking me to the Bible, it would be helpful if it could show me the chapters and verses that I have read and those that I still need to read, like in the app from the screenshot. This way, I can keep track of my reading more easily and accurately. 


https://github.com/searchwork/nwt-reading-contrib/assets/132299459/0fdd37a9-cd83-4f27-9e69-081042d04c7d


![IMG_4123](https://github.com/searchwork/nwt-reading-contrib/assets/132299459/e7830daf-a8c1-4a72-9f7b-fbe3d409b52d)

Thank you for your time and effort in developing this app!

**Comment by AngelAcuna2000** [comment]
Hi @wfleischer I was wondering if you have read my suggestions and what you thought about them.

**Comment by wfleischer** [comment]
Thank you @AngelAcuna2000 for this clarification. It seems like this is about a kind of “free-style” Bible reading, not following a certain reading plan. NWT Reading is designed to make following the reading plans as simple as possible. Adopting the scheme of the app you mentioned is not impossible, it would just mean to rework the way the app works or have to provided both ways within an app, which could be a bit confusing for the user.

So I am wondering if an option would be to just add the plan that you are following when reading the Bible to the app, if it is generally enough for a large public. Eventually the app could allow adding custom reading plans in the future. What do you think about these options?
